3/3.js

```diff
@@ -0,0 +1,222 @@
+        ingredients: ["chicken"],
+        type: "chicken",
+        available: false
+    },
+    {
+        id: 8,
+        name: "МакЧикен",
+        price: 4.3,
+        currency: "euro",
+        ingredients: ["chicken", "flour", "cheese", "sauce"],
+        type: "burger",
+        available: false
+    },
+
+
+
+]
+
+
+//1. Собрать в массив ингредиентов (без повторения).
+
+const allIngredients = data.reduce((acc, item) => {
+    const ingredients = item.ingredients
+    acc.push(ingredients)
+    return acc
+}, [])
+const allAccIngredients = allIngredients.join().split(',')
+const uniqIngredients = allAccIngredients.filter((item, index) => allAccIngredients.indexOf(item) === index);
+console.log(uniqIngredients);
+
+
+//2. Создать функцию, которая принимает массив продуктов и id, 
+//и возвращает продукт с таким же id.
+
+const findProduct = function(data, id) {;
+
+
+        const product = data.find(product => product.id === id)
+            //let product = cities.find((city) => {
+            //    return city.name === searchTerm).id}
+        return product
+    }
+    //console.log(findProduct(data, 8));
+
+
+//3. Создать массив с отсортированными продуктами по цене.
+
+
+function sortByPrice(data) {
+    return data.sort((a, b) => a.price - b.price);
+}
+//console.log(sortByPrice(data));
+
+
+//4. Сгруппировать продукты по типам. Создать объект, где ключ это тип, а значение - 
+//массив с продуктами.
+
+
+//5. Создать массив с продуктами, которые доступны.
+const newArray = data.filter(function(el) {
+        return el.available === false
+    })
+    //console.log(newArray);
+
+
+
+
+
+
+
+
+//6. Создать функцию, которая принимает массив продуктов и строку = название ингредиента, 
+//и возвращает массив с продуктами, где содержится такой ингредиент.
+const arrWithIngridient = (data, ingridient) => {
+        const newArr = data.filter(el => {
+            if (el.ingredients.indexOf(ingridient) != -1) {
+                return el
+            }
+
+        })
+        return newArr
+
+    }
+    //console.log(arrWithIngridient(data, 'potato'));
+    //console.log(arrWithIngridient(data, 'flour'));
+
+//7. Создать функцию, которая принимает массив продуктов и массив ингредиентов, 
+//и возвращает массив с продуктами, где содержатся такие ингредиенты.
+
+const arrWithIngridients = (data, ingredients) => {
+        const newArr = data.filter(el => {
+
+
+            let some = ingredients.every((item) => {
+                console.log(item)
+                return el.indexOf(item) !== -1;
+            })
+
+
+            if (some === true) {
+                console.log(234)
+                return el
+            }
+        })
+        return newArr
+
+    }
+    //console.log(arrWithIngridients(data, ["flour", "beef"]));
+
+//8. Создать функцию, которая принимает массив продуктов и цену, 
+//и возвращает массив продуктов, где цена продукта ниже или равна цене из второго аргумента функции.
+const arrWithPrice = (data, price) => {
```

Comment:
great! but it could be shorter
const filterProductsByPrice = (enterArray, price) => {
  return enterArray.filter((product) => product.price <= price);
};